corehq/ex-submodules/casexml/apps/phone/restore.py

```diff
@@ -79,6 +79,7 @@ def __init__(self, username=None, items=False):
         self.username = username
         self.items = items
         self.num_items = 0
+        self.finalized = False
```

Comment:
I considered it but finalizing closes the IO classes and attempt to write
to it will raise an error already.
On 13 Apr 2015 19:15, "Ben" <notifications@github.com> wrote:

> In corehq/ex-submodules/casexml/apps/phone/restore.py
> <https://github.com/dimagi/commcare-hq/pull/6301#discussion_r28257750>:
>
> > @@ -79,6 +79,7 @@ def __init__(self, username=None, items=False):
> >          self.username = username
> >          self.items = items
> >          self.num_items = 0
> > +        self.finalized = False
>
> I like the idea of having a finalized variable. one thing i was always
> concerned about was if a payload got accidentally written to after it was
> finalized. maybe we could even though an error __add__ or append
> functions if someone tries to modify the response after it was modified
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/6301/files#r28257750>.
>